test/gql/test_get.py

```diff
@@ -480,6 +480,188 @@ def test_build_near_image(self, mock_image_encoder_b64):
+        check_error_message(self, error, near_error_msg)
+
+    @patch("weaviate.gql.get.file_encoder_b64", side_effect=lambda x: "test_call")
+    def test_build_near_thermal(self, mock_file_encoder_b64):
+        """
+        Test the `with_near_object` method.
+        """
+
+        near_thermal = {"thermal": "test_thermal", "certainty": 0.55}
+
+        # valid calls
+        ## encode False
+        query = (
+            GetBuilder("Person", "name", None).with_near_thermal(near_thermal, encode=False).build()
+        )
+        self.assertEqual(
+            '{Get{Person(nearThermal: {thermal: "test_thermal" certainty: 0.55} ){name}}}', query
+        )
+        mock_file_encoder_b64.assert_not_called()
+
+        ## encode True
+        query = (
+            GetBuilder("Person", "name", None).with_near_thermal(near_thermal, encode=True).build()
+        )
+        self.assertEqual(
+            '{Get{Person(nearThermal: {thermal: "test_call" certainty: 0.55} ){name}}}', query
+        )
+        mock_file_encoder_b64.assert_called()
+
+        # invalid calls
+        near_error_msg = (
+            "Cannot use multiple 'near' filters, or a 'near' filter along with a 'ask' filter!"
+        )
+
+        near_text = {
+            "concepts": "computer",
+            "moveTo": {"concepts": ["science"], "force": 0.5},
+        }
+        with self.assertRaises(AttributeError) as error:
+            GetBuilder("Person", "name", None).with_near_text(near_text).with_near_thermal(
+                near_thermal
+            )
+        check_error_message(self, error, near_error_msg)
+
+    @patch("weaviate.gql.get.file_encoder_b64", side_effect=lambda x: "test_call")
+    def test_build_near_imu(self, mock_file_encoder_b64):
+        """
+        Test the `with_near_object` method.
+        """
+
+        near_imu = {"imu": "test_imu", "certainty": 0.55}
+
+        # valid calls
+        ## encode False
+        query = GetBuilder("Person", "name", None).with_near_imu(near_imu, encode=False).build()
+        self.assertEqual('{Get{Person(nearIMU: {imu: "test_imu" certainty: 0.55} ){name}}}', query)
+        mock_file_encoder_b64.assert_not_called()
+
+        ## encode True
+        query = GetBuilder("Person", "name", None).with_near_imu(near_imu, encode=True).build()
+        self.assertEqual('{Get{Person(nearIMU: {imu: "test_call" certainty: 0.55} ){name}}}', query)
+        mock_file_encoder_b64.assert_called()
+
+        # invalid calls
+        near_error_msg = (
+            "Cannot use multiple 'near' filters, or a 'near' filter along with a 'ask' filter!"
+        )
+
+        near_text = {
+            "concepts": "computer",
+            "moveTo": {"concepts": ["science"], "force": 0.5},
+        }
+        with self.assertRaises(AttributeError) as error:
+            GetBuilder("Person", "name", None).with_near_text(near_text).with_near_imu(near_imu)
+        check_error_message(self, error, near_error_msg)
```

Comment:
can you please parametrize all of these tests? That would save a lot of code and makes them easier to read and maintain

Reply:
You can't do exactly what we do elsewhere since these tests use the `unittest` Python library rather than `pytest`. However, `unittest` does have something similar. Have a look and let me know if it's good 😁

Reply:
yeah, there is a unittest parametrization or simply move the test out of the unittest class. All testa are run with pytest, eg adding pytest-tests works everywhere